packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -109,65 +109,63 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
-      // If match doesn't succeed, then give caller details about closest price periods.
+      // Reverse sort the historical price periods from newest to oldest. Find the first time period whose open time,
+      // minus the buffer, is before the target time. Use this matched period's close time.
       const before = this.historicalPricePeriods
         .slice()
         .reverse()
         .find((pricePeriod) => time >= pricePeriod.openTime - this.historicalTimestampBuffer);
-      const after = this.historicalPricePeriods.find(
-        (pricePeriod) => time < pricePeriod.closeTime + this.historicalTimestampBuffer
-      );
-
-      const format = (value: BN | null | undefined) => {
-        const unformattedValue = value && (this.invertPrice ? this._invertPriceSafely(value) : value);
-        if (!unformattedValue) return "[no value found]";
-        return formatFixed(unformattedValue.toString(), this.priceFeedDecimals);
-      };
-
-      throw new Error(`
-        Cryptowatch price feed ${this.uuid} didn't return an ohlc for time ${time}.
-        Closest price point before is close time: ${before?.closeTime} (- buffer of ${
-        this.historicalTimestampBuffer
-      } seconds), close price: ${format(before?.closePrice)}.
-        Closest price point after is open time: ${after?.openTime} (+ buffer of ${
-        this.historicalTimestampBuffer
-      } seconds), open price: ${format(after?.openPrice)}.
-        To see these prices, make a GET request to:
-        https://api.cryptowat.ch/markets/${this.exchange}/${this.pair}/ohlc?after=${
-        (before?.openTime || 0) + this.ohlcPeriod
-      }&before=${after?.closeTime}&periods=60
-      `);
-    }
 
-    const returnPrice = this.invertPrice ? this._invertPriceSafely(match.openPrice) : match.openPrice;
-    if (!returnPrice) throw new Error(`${this.uuid} -- invalid price returned`);
-    if (verbose) {
-      console.group(`\n(${this.exchange}:${this.pair}) Historical OHLC @ ${match.closeTime}`);
-      console.log(`- ✅ Open Price: ${formatFixed(returnPrice.toString(), this.priceFeedDecimals)}`);
-      console.log(
-        `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n- https://api.cryptowat.ch/markets/${this.exchange}/${this.pair}/ohlc?after=${match.closeTime}&before=${match.closeTime}&periods=60`
-      );
-      console.log(
-        '- This will return an OHLC data packet as "result", which contains in order: \n- [CloseTime, OpenPrice, HighPrice, LowPrice, ClosePrice, Volume, QuoteVolume].'
-      );
-      console.log(
-        "- We use the OpenPrice to compute the median. Note that you might need to invert the prices for certain identifiers like USDETH."
-      );
-      console.groupEnd();
+      // If the closest price period before the desired timestamp is within the historical timestamp buffer,
+      // then use its close price.
+      if (before && before.closeTime >= time - this.historicalTimestampBuffer) {
```

Comment:
If there is no match, then look through the price periods from newest to oldest, and return the first price period whose open time (minus the buffer) is before the target time